.buildkite/scripts/e2e/bootstrap.py

```diff
@@ -0,0 +1,150 @@
+"""
+E2E bootstrapping with Python script
+    - A script to resolve stack versions
+    - Download and spin up elastic-package
+    - Clone integrations repo and prepare packages
+    - When E2E finishes, teardown the stack
+"""
+import os
+import time
+from util import Util
+
+
+class Bootstrap:
+    ELASTIC_PACKAGE_DISTRO_URL = "https://api.github.com/repos/elastic/elastic-package/releases/latest"
+    LOGSTASH_CONTAINER_NAME = "elastic-package-stack-e2e-logstash-1"
+    PLUGIN_NAME = "logstash-filter-elastic_integration"
+
+    def __init__(self, stack_version: str, platform: str) -> None:
+        f"""
+        A constructor of the {Bootstrap}.
+
+        Args:
+            stack_version: An Elastic stack version where {Bootstrap} spins up with
+            platform: pass macos if you are on your dev Mac, otherwise linux 
+
+        Returns:
+            Validates platform, sets stack version and platform
+        """
+        self.stack_version = stack_version
+        self.__validate_platform(platform)
+        self.__set_distro(platform)
+
+    def __validate_platform(self, platform: str) -> None:
+        platforms = ["macos", "linux"]
+        if platform not in platforms:
+            raise ValueError(f"platform accepts {platforms} only")
+        self.platform = platform
+
+    def __set_distro(self, platform) -> None:
+        self.distro = "darwin_amd64.tar.gz" if platform == "macos" else "linux_amd64.tar.gz"
+
+    def __download_elastic_package(self) -> None:
+        response = Util.call_url_with_retry(self.ELASTIC_PACKAGE_DISTRO_URL)
+        release_info = response.json()
+
+        download_urls = [asset["browser_download_url"] for asset in release_info["assets"]]
+        download_url = [url for url in download_urls if self.distro in url][0]
+        if download_url.strip() == '':
+            raise Exception(f"Could not resolve elastic-package distro download URL, release info: {release_info}.")
+
+        file_name = "downloaded_elastic_package_" + self.distro
+        # downloading with `urllib3` gives a different size file which causes a corrupted file issue
+        Util.run_or_raise_error(["curl", "-o", file_name, "--retry", "5", "--retry-delay", "5", "-fSL", download_url],
+                                "Failed to download elastic-package")
+        print("elastic-package is successfully downloaded.")
+
+        # Extract the downloaded tar.gz file
+        Util.run_or_raise_error(["tar", "zxf", file_name],
+                                f"Error occurred while unzipping {file_name}")
+
+    def __make_elastic_package_global(self) -> None:
+        Util.run_or_raise_error(["sudo", "mv", "elastic-package", "/usr/local/bin"],
```

Comment:
root operation 😬 hope it works @dliappis

Reply:
if the paths are right it will work no worries